### ISSUE TITLE
Normalising YAML before diffing

### ISSUE THREAD

**Issue by sarahhodne** [databus23/helm-diff#257]
Sometimes when I upgrade Helm charts there's a change that doesn't really cause something to be changed, like an indentation change, or re-ordering the keys in a file. It would be nice if the YAML was "normalised" (ensure a consistent indentation level, sort all the keys by name) before diffing, to make it easier to see if there's actually a change being made.

**Comment by deiga** [comment]
Would that not cause a mismatch between what's actually in kube and what you are diffing with?

**Comment by lifelofranco** [comment]
This [issue](https://github.com/databus23/helm-diff/issues/254) might be related.

**Comment by schollii** [comment]
@deiga not if the values.yaml are semantically equivalent. Indentation, comments, blank lines, using or not using braces (recall yaml is superset of json so you can actually use braces to delimit blocks), and order of keys, do not change the actual configuration of the system. Changing the order of items in a list, renaming keys, and changing values of keys do change the configuration.

@sarahhodne, it would be useful if you posted an example of 2 values files that you consider identical but lead helm diff to show differences.

**Comment by sarahhodne** [comment]
The thing that made me create this ticket wasn't a change in the values file, but the chart we were using had changed (among a few other things) the indentation level, so a `Deployment` diff ended up something like this:

```diff
        containers:
-       - name: fluent-bit
+         - name: fluent-bit
```

There were also some key-reorderings, causing a diff like this:

```diff
+       volumes:
+         - name: config
+           configMap:
+             name: fluent-bit
        tolerations:
          - effect: NoSchedule
            key: pool-name
            operator: Exists
-       volumes:
-         - name: config
-           configMap:
-             name: fluent-bit
```

@deiga brings up a good point, but I think that's handled in #176 more than here, since `helm-diff` only diffs against Helm's internal state and not the actual live state (unlike Helm 3, which does a 3-way merge), this is already a problem orthogonal to this issue.

**Comment by davidholsgrove** [comment]
PR https://github.com/databus23/helm-diff/pull/273 if merged will close issue #257
